src/integritysetup/integritysetup-generator.c

```diff
@@ -0,0 +1,227 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <errno.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include "alloc-util.h"
+#include "fd-util.h"
+#include "fileio.h"
+#include "fstab-util.h"
+#include "generator.h"
+#include "hexdecoct.h"
+#include "id128-util.h"
+#include "main-func.h"
+#include "mkdir.h"
+#include "parse-util.h"
+#include "path-util.h"
+#include "proc-cmdline.h"
+#include "specifier.h"
+#include "string-util.h"
+#include "unit-name.h"
+
+static const char *arg_dest = NULL;
+static const char *arg_integritytab = NULL;
+static const char *arg_integritysetupbin = NULL;
+static char *arg_options = NULL;
+STATIC_DESTRUCTOR_REGISTER(arg_options, freep);
+
+static void option(
+                FILE *o,
+                char *token) {
+
+        char *save_ptr = NULL, *match = NULL;
+
+        assert(o);
+        assert(token);
+
+        match = strtok_r(token, "=", &save_ptr);
+        if (match) {
+                if (strlen(match) > 1)
+                        fprintf(o, "-");
+
+                fprintf(o, "-%s ", match);
+                match = strtok_r(NULL, "=", &save_ptr);
+                if (match)
+                        fprintf(o, "%s ", match);
+        }
+}
+
+static int options_string(
+                char *options,
+                char **options_formatted) {
+
+        char *buff = NULL, *save_ptr = NULL, *token = NULL;
+        size_t len = 0;
+        int r = 0;
```

Comment:
we don't usually initialize `r` on declaration

Reply:
> we don't usually initialize `r` on declaration

I see that this is indeed documented this way.  Does this just apply to the generic `r` return code or all stack variables?

Reply:
I removed the `r` initialization in latest push.